src/we-blog-smilies.1
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.32
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "WE-BLOG-SMILIES 1"
.TH WE-BLOG-SMILIES 1 "2012-07-23" "Version 0.9" "We-Blog Documentation"
.SH "NAME"
# we\-blog\-smilies \- Convert all smiley (emoticon) text tags to pictures
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
Bcat <file> | <we\-blog\-smilies>
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBwe-blog-smilies\fR converts all smiley (emoticon) text tags to pictures
.SH "BUGS"
.IX Header "BUGS"
To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/we\-blog/issues/>, or visit the
discussion group at <https://groups.google.com/d/forum/tonk\-we\-blog>.
.SH "COPYRIGHT"
.IX Header "COPYRIGHT"
Copyright (c) 2011\-2012 Ton Kersten
.PP
This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but \s-1WITHOUT\s0 \s-1ANY\s0 \s-1WARRANTY\s0;
without even the implied warranty of \s-1MERCHANTABILITY\s0 or \s-1FITNESS\s0 \s-1FOR\s0 A
\&\s-1PARTICULAR\s0 \s-1PURPOSE\s0.


ChangeLog
2012-07-10  Ton Kersten <we-blog@tonkersten.com>
  * Declared 0.8 stable
  * Switched to branch 0.9 for further development

2012-07-09  Ton Kersten <we-blog@tonkersten.com>
  * Changed a lot of things, but they all boil down to
    bugfixes and code reshuffeling

2011-10-07  Ton Kersten <we-blog@tonkersten.com>
  * Added 'alt' text for pages to show on mouse-over.

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * And switched to branch 0.8 for further development

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * Increased we-blog to version 0.7
  * Added 'smiley' converter
  * Published the project on GitHub

2011-08-15  Ton Kersten <we-blog@tonkersten.com>
  * Forked BlazeBlogger 1.2.0 to we-blog 0.5
    This because development of Blaze seems to be stalled

2011-02-18  Jaromir Hradilek <jhradilek@gmail.com>

  * INSTALL, Makefile, README, src/we-blog-add.pl, src/we-blog-config.pl,
  src/we-blog-edit.pl, src/we-blog-init.pl, src/we-blog-list.pl, src/we-blog-log.pl,
  src/we-blog-make.pl, src/we-blog-remove.pl, unix/we-blog.sh, unix/man/man1/we-blog.1,
  windows/we-blog.bat: Increased the version number.

  * unix/man/man1/we-blog.1: Updated the manual page for the command wrapper.

# $Id: ChangeLog 4 2012-07-10 09:34:11 tonk $


src/we-blog-smilies.pl
#!/usr/bin/env perl
# vi: set sw=4 ts=4 ai:
# $Id: we-blog-smilies.pl 3 2011-09-22 10:23:20 tonk $

# we-blog-smilies - Convert all smiley (emoticon) text tags to pictures

# Copyright (c) 2011-2012 Ton Kersten

# This program is free software: you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by the
# Free Software Foundation, version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but
# WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTA-
# BILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
# License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program. If not, see <http://www.gnu.org/licenses/>.

# $Author: Ton Kersten <we-blog@tonkersten.com> $
# $Date: 2011-09-22 10:26:18 +0200 (Thu, 22 Sep 2011) $
# $Fname: developer:.../src/we-blog-smilies.pl $
# $Revision: 3 $

use strict;
use warnings;
use Digest::MD5;
use File::Basename;
use File::Copy;
use File::Path;
use File::Spec::Functions;
use Getopt::Long;

# Set the library path and use our own module
use lib dirname($0);
use We;

# Global variables:
our $conf    = {};									# Configuration.

$conf = read_conf();

# Display usage information:
sub display_help {
    # Display the usage:
    print << "END_HELP";
Usage:  $NAME [-vh]
    $NAME -h|-v

    -h, --help                  display this help and exit
    -v, --version               display version information and exit
END_HELP

    # Return success:
    return 1;
}

# Set up the option parser:
Getopt::Long::Configure('no_auto_abbrev', 'no_ignore_case', 'bundling');

# Process command line options:
GetOptions(
    'help|h'         => sub { display_help();       exit 0; },
    'version|v'      => sub { display_version();    exit 0; },
);

my $smurl = $conf->{blog}->{smilies} || '/images/smilies';
$smurl = '<img class="smiley" alt="smiley" src="' . $smurl;

my %smilies = (
	':-{0,1}\)'		=> 'regular_smile.gif',		':-{0,1}D'	=> 'teeth_smile.gif',
	':-{0,1}O'		=> 'omg_smile.gif',			':-{0,1}P'	=> 'tongue_smile.gif',
	'\s+;-{0,1}\)'	=> 'wink_smile.gif',		':-{0,1}\('	=> 'sad_smile.gif',
	':-{0,1}S'		=> 'confused_smile.gif',	':-{0,1}\|'	=> 'what_smile.gif',
	':\'\('			=> 'cry_smile.gif',			':-{0,1}\$'	=> 'red_smile.gif',
	'\(H\)'			=> 'shades_smile.gif',		':-{0,1}\@'	=> 'angry_smile.gif',
	'\(A\)'			=> 'angel_smile.gif',		'\(6\)'		=> 'devil_smile.gif',
	':-{0,1}\#'		=> '47_47.gif',				'8o\|'		=> '48_48.gif',
	'8-{0,1}\|'		=> '49_49.gif',				'\^o\)'		=> '50_50.gif',
	':-{0,1}\*'		=> '51_51.gif',				'\+o\('		=> '52_52.gif',
	':\^\)'			=> '71_71.gif',				'\*-\)'		=> '72_72.gif',
	'\<:o\)'		=> '74_74.gif',				'8-\)'		=> '75_75.gif',
	'\|-{0,1}\)'	=> '77_77.gif',				'\(C\)'		=> 'coffee.gif',
	'\(Y\)'			=> 'thumbs_up.gif',			'\(N\)'		=> 'thumbs_down.gif',
	'\(B\)'			=> 'beer_mug.gif',			'\(D\)'		=> 'martini.gif',
	'\(X\)'			=> 'girl.gif',				'\(Z\)'		=> 'guy.gif',
	'\(\{\)'		=> 'guy_hug.gif',			'\(\}\)'	=> 'girl_hug.gif',
	'\:-{0,1}\['	=> 'bat.gif',				'\(\^\)'	=> 'cake.gif',
	'\(L\)'			=> 'heart.gif',				'\(U\)'		=> 'broken_heart.gif',
	'\(K\)'			=> 'kiss.gif',				'\(G\)'		=> 'present.gif',
	'\(F\)'			=> 'rose.gif',				'\(W\)'		=> 'wilted_rose.gif',
	'\(P\)'			=> 'camera.gif',			'\(\~\)'	=> 'film.gif',
	'\(\@\)'		=> 'cat.gif',				'\(\&\)'	=> 'dog.gif',
	'\(T\)'			=> 'phone.gif',				'\(I\)'		=> 'lightbulb.gif',
	'\(8\)'			=> 'note.gif',				'\(S\)'		=> 'moon.gif',
	'\(\*\)'		=> 'star.gif',				'\(E\)'		=> 'envelope.gif',
	'\(O\)'			=> 'clock.gif',				'\(sn\)'	=> '53_53.gif',
);

while ( <> )
{
	my $tag;
	foreach $tag (keys %smilies)
	{
		s!$tag!$smurl/$smilies{$tag}\" />!g;
	}
	print;
}
print "\n";

__END__

=head1 NAME

# we-blog-smilies - Convert all smiley (emoticon) text tags to pictures

=head1 SYNOPSIS

Bcat <file> | <we-blog-smilies>

=head1 DESCRIPTION

B<we-blog-smilies> converts all smiley (emoticon) text tags to pictures

=head1 BUGS

To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/we-blog/issues/>, or visit the
discussion group at <https://groups.google.com/d/forum/tonk-we-blog>.

=head1 COPYRIGHT

Copyright (c) 2011-2012 Ton Kersten

This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

=cut
